resolwe_bio/processes/reads_processing/trimmomatic.yml

```diff
@@ -131,193 +149,281 @@
-      type: data:seq:nucleotide
-      required: false
     - name: illuminaclip
       label: Illumincaclip
       group:
+        - name: adapters
+          label: Adapter sequences
+          type: data:seq:nucleotide
+          required: false
+          description: >
+            Adapter sequence in FASTA format that will be removed from the read.
+            This field as well as 'Seed mismathes', 'Simple clip threshold' and 'Palindrome clip threshold' parameters are
+            needed to perform Illuminacliping. 'Minimum adapter length' and 'Keep both reads' are optional parameters.
         - name: seed_mismatches
           label: Seed mismatches
           type: basic:integer
           required: false
           description: >
-             Specifies the maximum mismatch count which will still allow a full match to be performed.
+            Specifies the maximum mismatch count which will still allow a full match to be performed.
+            This field as well as 'Adapter sequence', 'Simple clip threshold' and 'Palindrome clip threshold' parameters are
+            needed to perform Illuminacliping.
         - name: simple_clip_threshold
           label: Simple clip threshold
           type: basic:integer
           required: false
           description: >
             Specifies how accurate the match between any adapter etc. sequence must be against a read.
+            This field as well as 'Adapter sequence', 'Seed mismatches' and 'Palindrome clip threshold' parameters are
+            needed to perform Illuminacliping
         - name: palindrome_clip_threshold
           label: Palindrome clip threshold
           type: basic:integer
           required: false
           description: >
             Specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment.
+            This field as well as 'Adapter sequence', 'Simple clip threshold' and 'Seed mismatches' parameters are
+            needed to perform Illuminacliping
         - name: min_adapter_length
-          label: Min adapter length
+          label: Minimum adapter length
           type: basic:integer
+          disabled: >
+            !illuminaclip.seed_mismatches &&
+            !illuminaclip.simple_clip_threshold &&
+            !illuminaclip.palindrome_clip_threshold
           required: false
           description: >
-             In addition to the alignment score, palindrome mode can verify that a minimum length of adapter has been detected. If unspecified, this defaults to 8 bases, for historical reasons. However, since palindrome mode has a very low false positive rate, this can be safely reduced, even down to 1, to allow shorter adapter fragments to be removed.
+             In addition to the alignment score, palindrome mode can verify that a minimum length of adapter has been
+             detected. If unspecified, this defaults to 8 bases, for historical reasons. However, since palindrome mode
```

Comment:
If this input defaults to 8 bases, the `default` attribute of this input should be set. Then, `required: false` is not needed anymore as this input will always be set. Adjust warning messages accordingly.